framework/managers/MqttConnectionManager.cpp

```diff
@@ -61,8 +61,12 @@ void MqttConnectionManager::onDisconnected(TcpClient &client, bool flag) {
         LOG.log("Unreachable");
     }
 
-    this->state.set(State::DISCONNECTED);
-    this->reconnectTimer.start();
+    if (this->getState() != State::CONNECTING) {
```

Comment:
We should ignore disconnects while being disconnected, too.

Reply:
Hm, while being disconnected that would only restart the timer? So it shouldn't be a problem but ok

Reply:
And should I check for flag too? If the broker is not reachable it should be false and the state should be CONNECTING, hmm

Reply:
I think checking for `== CONECTED` should be fine.